src/__tests__/shared/components/input/TestInputWithLabelFormGroup.test.tsx

```diff
@@ -8,92 +8,86 @@ describe('text input with label form group', () => {
     })
 
     it('should render a text field', () => {
-      const expectedName = 'test'
-      const wrapper = shallow(
+      const expectedLabel = 'test'
+      render(
         <TextInputWithLabelFormGroup
-          name={expectedName}
-          label="test"
+          name="test"
+          label={expectedLabel}
           value=""
           isEditable
           onChange={jest.fn()}
         />,
       )
 
-      const input = wrapper.find(TextInput)
-      expect(input).toHaveLength(1)
+      expect(screen.getByLabelText(expectedLabel)).toBeInTheDocument()
     })
 
     it('should render disabled is isDisable disabled is true', () => {
-      const expectedName = 'test'
-      const wrapper = shallow(
+      const expectedLabel = 'test'
+      render(
         <TextInputWithLabelFormGroup
-          name={expectedName}
-          label="test"
+          name="test"
+          label={expectedLabel}
           value=""
           isEditable={false}
           onChange={jest.fn()}
         />,
       )
 
-      const input = wrapper.find(TextInput)
-      expect(input).toHaveLength(1)
-      expect(input.prop('disabled')).toBeTruthy()
+      expect(screen.getByLabelText(expectedLabel)).toBeDisabled()
     })
 
     it('should render the proper value', () => {
-      const expectedName = 'test'
+      const expectedLabel = 'test'
       const expectedValue = 'expected value'
-      const wrapper = shallow(
+      render(
         <TextInputWithLabelFormGroup
-          name={expectedName}
-          label="test"
+          name="test"
+          label={expectedLabel}
           value={expectedValue}
           isEditable={false}
           onChange={jest.fn()}
         />,
       )
 
-      const input = wrapper.find(TextInput)
-      expect(input).toHaveLength(1)
-      expect(input.prop('value')).toEqual(expectedValue)
+      expect(screen.getByLabelText(expectedLabel)).toHaveAttribute('value', expectedValue)
     })
   })
 
   describe('change handler', () => {
     it('should call the change handler on change', () => {
-      const expectedName = 'test'
+      const expectedLabel = 'test'
       const expectedValue = 'expected value'
       const handler = jest.fn()
-      const wrapper = shallow(
+      render(
         <TextInputWithLabelFormGroup
-          name={expectedName}
-          label="test"
-          value={expectedValue}
-          isEditable={false}
+          name="test"
+          label={expectedLabel}
+          value=""
+          isEditable
           onChange={handler}
         />,
       )
 
-      const input = wrapper.find(TextInput)
-      input.simulate('change')
-      expect(handler).toHaveBeenCalledTimes(1)
+      const input = screen.getByLabelText(expectedLabel)
+      userEvent.type(input, expectedValue)
+      expect(input).toHaveValue(expectedValue)
+      expect(handler).toHaveBeenCalledTimes(expectedValue.length)
```

Comment:
I handled this [a bit differently](https://github.com/kcdraidgroup/hospitalrun-frontend/blob/master/src/__tests__/shared/components/input/TextFieldWithLabelFormGroup.test.tsx#L97) but I don't mind your approach here either.

Reply:
Should I change it so it will be similar to yours? Will we come back again to that strange onChange behaviour?

Reply:
Ideally, we don't want to test code implementation if we can, if we can check if something in the DOM changed then the  checking if it was called because unnecessary (function called is an implementation detail IMO)

Reply:
I think this case is slightly different. We want to ensure the provided callback is called when the input is changed, not just that typing into it updates the DOM. 

A mock function to capture the calls is ok, but this particular `onChange` is sending weird events that I wasn't expecting (a single character at a time), so the assertion looks a bit odd.

@ziedtouibi My version does also check the _correct_ characters come through, so I think it's slightly better than just checking the number of calls, but yours is closer to the original test, so I'm happy either way. We're planning on doing second pass anyway to "improve" the tests so we should try not to get too hung up on the little things on this first pass (there's way too much to do already).

Reply:
> I think this case is slightly different. We want to ensure the provided callback is called when the input is changed, not just that typing into it updates the DOM.

Generally from an RTL standpoint asserting the code including a callback like this has fired is an implementation detail, if the code works the related behavior should be reflected on the DOM. 

I do agree that it is closer to the original test though and minor detail. 

> (there's way too much to do already)
Something I 100% agree with and say myself lol